modules/nf-core/segemehl/align/main.nf

```diff
@@ -0,0 +1,56 @@
+process SEGEMEHL_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::segemehl=0.3.4" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/segemehl:0.3.4--hc2ea5fd_5':
+        'quay.io/biocontainers/segemehl:0.3.4--hc2ea5fd_5' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path(fasta)
+    path(index)
+
+    output:
+    tuple val(meta), path("${meta.id}/*"), emit: results
+    path "versions.yml"                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reads = meta.single_end ? "-q ${reads}" : "-q ${reads[0]} -p ${reads[1]}"
+    def suffix = args.contains("-b") ? "bam" : "sam"
```

Comment:
```suggestion
    def suffix = (args.contains("-b ") || args.contains("--bamabafixoida")) ? 'bam' : 'sam'
```
Also, I ran some tests locally and '--bam' didn't work for me. But then I am not familiar with the tool so I don't if I screwed up my tests. Does `--bam` work for you?

Reply:
Also, I think it would be cool to include tests to make sure that the prefix and output format are set as expected. I am thinking something like [this](https://github.com/nf-core/modules/blob/master/tests/modules/nf-core/hmmer/hmmalign/main.nf).

Reply:
@ramprasadn I've added tests for bam file outputs that seem to be working well. 

Whilst adding a test for the prefix, I learned something interesting. I had to remove the `def` from the script preamble for nextflow to recognize `prefix` in the output declaration. The same behavior is observed here in `qualimap`:

https://github.com/nf-core/modules/blob/07c9080d7b66a325e8e67d9c57e10083ca8b9db4/modules/nf-core/qualimap/bamqc/main.nf#L15-L23

When using the canonical `def prefix = task.ext.prefix ?: "${meta.id}"`, I got the error `Missing output file(s) null/* expected by process test_segemehl_align_prefix:SEGEMEHL_PREFIX (test)`.

Reply:
That's right. I remember reading about this on nf-core slack a while ago. And my takeaway from that thread was that by adding def keyword in front of a variable we keep it in local scope. So if we want to access it outside of the script section then we need to omit `def`.